docs_src/design/adr/device-service/0021-Profile-Changes.md

```diff
@@ -0,0 +1,116 @@
+| Manufacturer | string | Anytime | As this property does not affect any functionality other than profile query, this property can be changed at any time – to include setting the string to an empty string. |
+| Description | string | Anytime | As this property does not affect any functionality other than profile query, this property can be changed at any time – to include setting the string to an empty string. |
+| Model | string | Anytime | As this property does not affect any functionality other than profile query, this property can be changed at any time – to include setting the string to an empty string. |
+| Labels | string array | Anytime | As this property does not affect any functionality other than profile query, this property can be changed at any time – to include adding, removing or modifying any of the strings in the array. |
+| Device Resources | array | Situational | When the device resource is not associated to any reading or event, the device resource list can be modified (i.e., adding or removing the associated device resources).  Once the device resource is associated to a reading or event (directly or indirectly via device command), only new resources can be added to the array.  Deleting is prohibited.|
+| Device Commands | array | Situational | When the device command is not associated to any reading or event, the device commands list can be modified (i.e., adding or removing the associated device commands).  Once the device command is associated to a reading or event, only new commands can be added to the array.  Deleting is prohibited.  Device commands are optional and so the array can always be empty. |
+| *Device Resource Properties* |||
+| Description | string | Anytime | As this property does not affect any functionality, this property can be changed at any time – to include setting the string to an empty string. |
+| Name | string | Situational | When the device profile is not associated to any reading or event, this field can be modified.  It can never be blank and must be unique across the entire profile. |
+| IsHidden | bool | Anytime | This field allows the device resource to be exposed (default) via the command service.  This change does not affect the behavior but only visibility.  This field is optional but is false by default when not provided. |
+| Tag | string | Situational | As this property does not affect any functionality, this property can be changed at any time – to include setting the string to an empty string. |
+| Properties | Resource Properties | Situational | When the device resource (directly or indirectly through device command) is not associated to any reading or event, this field can be modified.  Because the ValueType and ReadWrite properties of this field are required, it can never be empty. |
```

Comment:
`MediaType`?

Reply:
I also would argue that since some of the properties are actually used to calculate the value of readings, changing some of them (e.g. `Mask`, `Shift`, `Scale`, `Offset`, `Base`) could mean two readings with identical values might've been calculated differently.

Reply:
Because there seems to still be some discussion here, I've left this one open to discuss in the Core WG.  Also, since we don't enumerate Properties here, by fact that some are Situation (whether ValueType or others) then Properties as a whole are situation.  Or we have to list out all the properties and say which are allowed to change and which are not.

Reply:
Also please note that modifications to `ValueType` don't actually have an impact on existing readings because readings have a local `ValueType` which is set from the device resource `ValueType` when the reading is created. The same applies to `MediaType`. Most of the other properties (including `Units`) can only ever be accessed via the device resource.

IMHOP the simplest approach would be all or nothing (i.e. if the resource has been used/associated w/an event or reading**) don't allow any properties to be changed other than `ReadWrite` and `Assertion`. One could also argue that `Minimum`, `Maximum`, and `DefaultValue` also fall into this category.

** Note, I realize the reason we don't want to allow changes in this idea that we don't want referential integrity issues, however I'll point out that our Core Data persistence is (a) optional and (b) meant to be cleaned up periodically, so I'm not sure what problem we're trying to solve by ensuring that no changes occur to the device resource if these short-term readings are available from Core Data. Perhaps as part of our _Units of Measure ADR_, we also propose that units get added to the readings, which means our readings are self-describing (i.e. they'd contain `ValueType`, `Units`, and `MediaType` if the reading is binary). All of the other properties have to do with how readings are created (e.g. `Base`, `Mask`, `DefaultValue`, ...), how they can used (e.g. `ReadWrite`), or whether they trigger an error (e.g. `Assertion`).

Reply:
> so I'm not sure what problem we're trying to solve by ensuring that no changes occur to the device resource if these short-term readings are available from Core Data.

@tonyespy , I have been thinking the same. If Core Data isn't persisting or Events/Readings have be removed then none of the rules with respected to associated Events/Readings matter...

Reply:
Per Core WG meeting of 1/13/21 - changing to allow change to profile anytime the profile is not attached to a device or provision watcher.  Event and reading association doesn't matter since these objects carry the profile, resource name and value type already and because these are short lived objects.